config/initializers/zeitwerk.rb

```diff
@@ -0,0 +1,15 @@
+if Rails.autoloaders.main
+  Zeitwerk::Loader.default_logger = method(:puts)
+  Rails.autoloaders.main.logger = Logger.new(STDOUT)
+
+  # These specific directories are for code organization, not namespacing:
+  Rails.autoloaders.main.collapse(Rails.root.join("lib/manageiq/reporting/charting"))
+  Rails.autoloaders.main.collapse(Rails.root.join("lib/ansible/runner/credential"))
+
+  Rails.autoloaders.main.ignore(%w[
+    /Users/joerafaniello/.gem/ruby/3.0.6/bundler/gems/manageiq-providers-nuage-d87b99b29624/lib/manageiq-providers-nuage.rb
+  ])
+
+  # requires qpid, which isn't available on my system
```

Comment:
Is there a way to handle these "optional" requires?

Reply:
Good question.  I still need to figure this out.

Reply:
For now I just programmatically build a path to the file.

```ruby
  # requires qpid, which is an optional dependency because LOL mac
  message_handler_path = Pathname.new(Vmdb::Plugins.paths[ManageIQ::Providers::Nuage::Engine]).join("app/models/manageiq/providers/nuage/network_manager/event_catcher/messaging_handler.rb")
  Rails.autoloaders.main.ignore(message_handler_path)
```